### PR TITLE
Simplify and speed up "kpt live" e2e tests

### PR DESCRIPTION
**Pull request by mortent**
Simplifies the test config and schedules tests to avoid having to recreate the kind cluster (which is very slow).

### CODE REVIEW

File: e2e/testdata/live-apply/apply-depends-on/config.yaml [hunk 1]
```diff
@@ -12,20 +12,14 @@
...
-      {"eventType":"resourceApplied","group":"apps","kind":"Deployment","name":"second-nginx","namespace":"apply-depends-on","operation":"Created","timestamp":"<TIMESTAMP>","type":"apply"}
-      {"configuredCount":0,"count":1,"createdCount":1,"eventType":"completed","failedCount":0,"serverSideCount":0,"timestamp":"<TIMESTAMP>","type":"apply","unchangedCount":0}
+parallel: true
+
+stdOut: |
+  deployment.apps/first-nginx created
+  1 resource(s) applied. 1 created, 0 unchanged, 0 configured, 0 failed
+  deployment.apps/second-nginx created
+  1 resource(s) applied. 1 created, 0 unchanged, 0 configured, 0 failed
+
```
Comment (droot) [inline e2e/testdata/live-apply/apply-depends-on/config.yaml:22]: any reason for taking out the json output ?

File: e2e/testdata/live-apply/json-output/config.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+parallel: true
+kptArgs:
+  - "--output=json"
+stdOut: |
```
Comment (droot) [inline e2e/testdata/live-apply/json-output/config.yaml:18]: ignore my previous comment. It looks like you are adding an explicit test for testing json-output.

File: pkg/test/live/config.go [hunk 3]
```diff
@@ -26,33 +26,28 @@ type TestCaseConfig struct {
...
 	Inventory []InventoryEntry `yaml:"inventory,omitempty"`
 
-	// RequiresCleanCluster tells the test framework that a new cluster must
-	// be created for running this test.
-	RequiresCleanCluster bool `yaml:"requiresCleanCluster,omitempty"`
-
 	// PreinstallResourceGroup causes the test framework to verify that the
 	// ResourceGroup CRD is available in the cluster before running the test.
-	PreinstallResourceGroup bool `yaml:"preinstallResourceGroup,omitempty"`
+	NoResourceGroup bool `yaml:"noResourceGroup,omitempty"`
```
Comment (droot) [inline pkg/test/live/config.go:41]: nit: The comment is outdated.

**Review by droot — APPROVED** [review]
minor comments.

File: pkg/test/live/config.go (hunk 3 above)
Comment (mortent) [reply pkg/test/live/config.go:41]: Fixed.

File: e2e/testdata/live-apply/json-output/config.yaml (hunk 2 above)
Comment (mortent) [reply e2e/testdata/live-apply/json-output/config.yaml:18]: Yeah, having the same test looking at multiple output formats turned out to be difficult to manage. So removed that functionality and instead have separate tests for alternative output formats.

**Comment by yuwenma** [comment]
/lgtm

Thanks, @droot !

**Comment by yuwenma** [comment]
> Simplifies the test config and schedules tests to avoid having to recreate the kind cluster (which is very slow).

really like

**Pull request merged**